deploy/cdkpipeline/cdkpipeline.py

```diff
@@ -0,0 +1,147 @@
+                ],
+                action_name = "Synth",
+                project_name = "cdk_synth",
+                subdirectory = "deploy"
+            )
+
+        )
+            
+
+        # Deploy infrastructure
+        pipeline.add_application_stage(
+            SampleSheetCheckStage(
+                self,
+                "SampleSheetCheckFrontEndStage",
+            )
+        )
+
+        react_build_stage = pipeline.add_stage(
+            stage_name="ReactBuild",
+        )
+        front_end_bucket_name=ssm.StringParameter.from_string_parameter_attributes(
+            self,
+            "FrontEndBucketName",
+            parameter_name="/sscheck/bucket_name"
+        ).string_value
+
+        front_end_bucket_arn = s3.Bucket.from_bucket_name(
+            self,
+            "FrontEndBucket",
+            bucket_name=front_end_bucket_name
+        ).bucket_arn
+
+        react_build_stage.add_actions(
+            pipelines.ShellScriptAction(
+                action_name = "BuildScript",
+                commands = [
+                    "export REACT_APP_BUCKET_NAME=$(aws ssm get-parameter --name '/sscheck/bucket_name' | jq -r .Parameter.Value)",
+                    "export REACT_APP_LAMBDA_API_DOMAIN=$(aws ssm get-parameter --name '/sscheck/lambda-api-domain' | jq -r .Parameter.Value)",
+                    "export REACT_APP_STAGE=$(aws ssm get-parameter --name '/sscheck/stage' | jq -r .Parameter.Value)",
+                    "export REACT_APP_REGION=ap-southeast-2",
+
+                    "export REACT_APP_COG_USER_POOL_ID=$(aws ssm get-parameter --name '/data_portal/client/cog_user_pool_id' | jq -r .Parameter.Value)",
+                    "export REACT_APP_COG_APP_CLIENT_ID_STAGE=$(aws ssm get-parameter --name '/sscheck/client/cog_app_client_id_stage' | jq -r .Parameter.Value)",
+
+                    "export REACT_APP_OAUTH_DOMAIN=$(aws ssm get-parameter --name '/data_portal/client/oauth_domain' | jq -r .Parameter.Value)",
+                    "export REACT_APP_OAUTH_REDIRECT_IN_STAGE=$(aws ssm get-parameter --name '/sscheck/client/oauth_redirect_in_stage' | jq -r .Parameter.Value)",
+                    "export REACT_APP_OAUTH_REDIRECT_OUT_STAGE==$(aws ssm get-parameter --name '/sscheck/client/oauth_redirect_out_stage' | jq -r .Parameter.Value)",
+                    "env | grep REACT",
+                    "npm i react-scripts",
+                    "npm run build",
+                    "npm run deploy"
+                ],
+                additional_artifacts = [source_artifact],
+                role_policy_statements = [
+                    iam.PolicyStatement(
+                        actions=["ssm:GetParameter"],
+                        effect=iam.Effect.ALLOW,
+                        resources=[
+                            "arn:aws:ssm:ap-southeast-2:843407916570:parameter/sscheck/*",
+                            "arn:aws:ssm:ap-southeast-2:843407916570:parameter/data_portal/*"
```

Comment:
Not major, but in the interest of least privilege principle:
Portal SSM parameter pattern could be restricted further to `data_portal/client/*`?

Reply:
Yup agree, added on `a6afdcc`